lib/functions/transferpartscreate.js

```diff
@@ -57,23 +57,24 @@ class CreateTransferParts extends AsyncGeneratorFunction {
         for await (const transferAsset of transferAssets) {
             const contentLength = transferAsset.metadata.contentLength;
             if (transferAsset.acceptRanges && transferAsset.multipartTarget) {
-                const { targetUrls, minPartSize, maxPartSize, headers } = transferAsset.multipartTarget;
+                const { headers: sourceHeaders } = transferAsset.source;
+                const { targetUrls, minPartSize, maxPartSize, headers: targetHeaders } = transferAsset.multipartTarget;
                 const partSize = calculatePartSize(targetUrls.length, contentLength, minPartSize, maxPartSize, this.preferredPartSize);
                 let idx = 0;
                 for (const range of generatePartRanges(contentLength, partSize)) {
-                    yield new TransferPart(transferAsset, [targetUrls[idx]], range, headers);
+                    yield new TransferPart(transferAsset, [targetUrls[idx]], range, sourceHeaders, targetHeaders);
                     ++idx;
                 }   
             } else if (transferAsset.acceptRanges && isFileProtocol(transferAsset.target.url)) {
                 const targetUrls = [transferAsset.target.url];
                 const partSize = this.preferredPartSize || DEFAULT_FILE_TARGET_PART_SIZE;
                 for (const range of generatePartRanges(contentLength, partSize)) {
-                    yield new TransferPart(transferAsset, targetUrls, range);
+                    yield new TransferPart(transferAsset, targetUrls, range, transferAsset.source.headers, transferAsset.target.headers);
                 }
             } else {
                 const targetUrls = [transferAsset.target.url];
                 const contentRange = new DRange(0, contentLength - 1);
-                yield new TransferPart(transferAsset, targetUrls, contentRange, transferAsset.target.headers);
+                yield new TransferPart(transferAsset, targetUrls, contentRange, transferAsset.source.headers, transferAsset.target.headers);
```

Comment:
With the change above to make `TransferPart` have a convenience getter to get `transferAsset.source.headers`, this can be removed.